plugins/deployment_manager/DeploymentManager.ts

```diff
@@ -133,46 +146,105 @@ export class DeploymentManager {
     );
   }
 
+  /* Unconditionally casts a contract as the given artifact type, without caching */
+  async cast<C extends Contract>(address: string, artifact: string): Promise<C> {
+    const buildFile = await readContract(this.cache, this.hre, artifact, address, true);
+    return getEthersContract<C>(address, buildFile, this.hre);
+  }
+
+  /* Conditionally clones a contract with its alias from a given network to this deployment */
+  async clone<C extends Contract>(
+    alias: Alias,
+    address: string,
+    deployArgs: any[],
+    fromNetwork?: string,
+    force?: boolean,
+    retries?: number
+  ): Promise<C> {
+    const maybeExisting: C = await this.contract(alias);
+    if (!maybeExisting || force) {
+      // NB: might use an override alias on deployOpts, but it doesn't invalidate cache...
+      const buildFile = await this.import(address, fromNetwork);
+      const contract: C = await this._deployBuild(buildFile, deployArgs, retries);
+      await this.putAlias(alias, contract.address);
+      return contract;
+    }
+    return maybeExisting;
+  }
+
+  /* Conditionally deploy a contract with its alias if it does not exist, or if forced */
+  async deploy<C extends Contract, DeployArgs extends Array<any>>(
```

Comment:
This is more like "I'm ron burgundy?", I mean `deploy?`. or really `fetchOrDeploy`

Reply:
`idempotent` deploy 😬

Reply:
yeah `idempotentDeploy` might be nice, since it's a bit more descriptive to what this is?